src/diff.jl

```diff
@@ -33,17 +34,17 @@ struct Differential <: Operator
     x
     Differential(x) = new(value(x))
 end
-(D::Differential)(x) = Term{symtype(x)}(D, [x])
-(D::Differential)(x::Num) = Num(D(value(x)))
-(D::Differential)(x::Complex{Num}) = wrap(ComplexTerm{Real}(D(unwrap(real(x))), D(unwrap(imag(x)))))
+(D::Operator)(x) = Term{symtype(x)}(D, [x])
+(D::Operator)(x::Num) = Num(D(value(x)))
+(D::Operator)(x::Complex{Num}) = wrap(ComplexTerm{Real}(D(unwrap(real(x))), D(unwrap(imag(x)))))
 SymbolicUtils.promote_symtype(::Differential, x) = x
 
 is_derivative(x) = istree(x) ? operation(x) isa Differential : false
 
-Base.:*(D1, D2::Differential) = D1 ∘ D2
-Base.:*(D1::Differential, D2) = D1 ∘ D2
-Base.:*(D1::Differential, D2::Differential) = D1 ∘ D2
-Base.:^(D::Differential, n::Integer) = _repeat_apply(D, n)
+Base.:*(D1, D2::Operator) = D1 ∘ D2
+Base.:*(D1::Operator, D2) = D1 ∘ D2
```

Comment:
I'm wondering why we have these methods @ChrisRackauckas ? this does not make sense in general,  only maybe for 2 operators

Reply:
What's a case where they don't make sense?